cluster-autoscaler/core/scaledown/actuation/delete_in_batch_test.go

```diff
@@ -168,15 +169,18 @@ func TestRemove(t *testing.T) {
 				t.Fatalf("Couldn't set up autoscaling context: %v", err)
 			}
 
+			scaleStateNotifier := observers.NewNGChangeObserversList()
+			scaleStateNotifier.Register(clusterStateRegistry)
```

Comment:
I guess you are asking why `clusterStateRegistry` was originally added in this test. Setting it to `nil` leads to a test failure because of nil pointer error:
```
$ go test -run TestRemove
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1c4d633]

goroutine 149 [running]:
k8s.io/autoscaler/cluster-autoscaler/clusterstate.(*ClusterStateRegistry).RegisterScaleDown(0x0, 0xc0004e6000)
        /home/suraj/sandbox/ca-master/cluster-autoscaler/clusterstate/clusterstate.go:244 +0x33
k8s.io/autoscaler/cluster-autoscaler/core/scaledown/actuation.RegisterAndRecordSuccessfulScaleDownEvent(0xc000436d80, 0xc000463a10?, 0xc00035fb80, {0x2607948?, 0xc0001a2c00}, 0x1, 0x0?)
        /home/suraj/sandbox/ca-master/cluster-autoscaler/core/scaledown/actuation/delete_in_batch.go:198 +0x1e5
k8s.io/autoscaler/cluster-autoscaler/core/scaledown/actuation.(*NodeDeletionBatcher).remove.func1({0xc0000532c0?, 0x5, 0x0?}, 0x0?)
        /home/suraj/sandbox/ca-master/cluster-autoscaler/core/scaledown/actuation/delete_in_batch.go:139 +0x19f
created by k8s.io/autoscaler/cluster-autoscaler/core/scaledown/actuation.(*NodeDeletionBatcher).remove in goroutine 147
        /home/suraj/sandbox/ca-master/cluster-autoscaler/core/scaledown/actuation/delete_in_batch.go:130 +0x345
exit status 2
FAIL    k8s.io/autoscaler/cluster-autoscaler/core/scaledown/actuation   0.108s
```

[Call to register scale down panics](https://github.com/kubernetes/autoscaler/blob/57f2814267399e0900a70d567a7813089a3cfa5b/cluster-autoscaler/core/scaledown/actuation/delete_in_batch.go#L198-L203) because the code tries to call a function on `clusterStateRegistry` (aka `csr`) but it is nil.

Reply:
But otherwise agree with removing `clusterStateRegistry`. We can use an empty `scaleStateNotifier` with nothing to notify to. 

I confirmed the tests pass:
```
$ go test -run TestRemove
E1220 23:18:09.717346   64243 delete_in_batch.go:165] Couldn't determine node "ng-node-0" readiness while scaling down - assuming unready: readiness information not found
E1220 23:18:09.717389   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
E1220 23:18:09.717395   64243 delete_in_batch.go:165] Couldn't determine node "ng-node-1" readiness while scaling down - assuming unready: readiness information not found
E1220 23:18:09.717399   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
E1220 23:18:09.717402   64243 delete_in_batch.go:165] Couldn't determine node "ng-node-2" readiness while scaling down - assuming unready: readiness information not found
E1220 23:18:09.717406   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
E1220 23:18:09.717409   64243 delete_in_batch.go:165] Couldn't determine node "ng-node-3" readiness while scaling down - assuming unready: readiness information not found
E1220 23:18:09.717413   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
E1220 23:18:09.717416   64243 delete_in_batch.go:165] Couldn't determine node "ng-node-4" readiness while scaling down - assuming unready: readiness information not found
E1220 23:18:09.717419   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
E1220 23:18:09.717700   64243 delete_in_batch.go:187] Scale-down: couldn't delete node "ng-node-0" with drain, , status error: failed to delete nodes from group ng: SIMULATED ERROR: won't remove node
E1220 23:18:09.717734   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
E1220 23:18:09.717738   64243 delete_in_batch.go:187] Scale-down: couldn't delete node "ng-node-1" with drain, , status error: failed to delete nodes from group ng: SIMULATED ERROR: won't remove node
E1220 23:18:09.717746   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
E1220 23:18:09.717749   64243 delete_in_batch.go:187] Scale-down: couldn't delete node "ng-node-2" with drain, , status error: failed to delete nodes from group ng: SIMULATED ERROR: won't remove node
E1220 23:18:09.717757   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
PASS
E1220 23:18:09.717760   64243 delete_in_batch.go:187] Scale-down: couldn't delete node "ng-node-3" with drain, , status error: failed to delete nodes from group ng: SIMULATED ERROR: won't remove node
E1220 23:18:09.717768   64243 nodedeletiontracker.go:93] This should never happen, counter for ng in NodeDeletionTracker wasn't found
ok      k8s.io/autoscaler/cluster-autoscaler/core/scaledown/actuation   0.021s
```

Removed in https://github.com/kubernetes/autoscaler/pull/5729/commits/0b77571b20b20c1fce880004d754be89edc9b6b4